src/client/power-pages/create/Contentsnippet.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as nls from "vscode-nls";
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+// const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import * as vscode from "vscode";
+import {
+    createFileWatcher,
+    createRecord,
+    formatFileName,
+    formatFolderName,
+    isNullOrEmpty,
+} from "./utils/CommonUtils";
+import { QuickPickItem } from "vscode";
+import { MultiStepInput } from "./utils/MultiStepInput";
+import path from "path";
+import { statSync } from "fs";
+import { Tables, YoSubGenerator } from "./constants";
+
+export const createContentSnippet = async (
+    context: vscode.ExtensionContext,
+    selectedWorkspaceFolder: string | undefined,
+    yoGenPath: string | null
+): Promise<void> => {
+    try {
+        if (selectedWorkspaceFolder) {
+            const { contentSnippetName, contentSnippetType } =
+                await getContentSnippetInputs(selectedWorkspaceFolder);
+
+            if (!isNullOrEmpty(contentSnippetName)) {
+                const folder = formatFolderName(contentSnippetName);
+                const file = formatFileName(contentSnippetName);
+
+                const watcherPattern = path.join(
+                    Tables.CONTENT_SNIPPET,
+                    folder,
+                    `${file}.*.contentsnippet.yml`
+                );
+                const watcher = createFileWatcher(
+                    context,
+                    selectedWorkspaceFolder,
+                    watcherPattern
+                );
+
+                const portalDir = selectedWorkspaceFolder;
+
+                const command = `"${yoGenPath}" ${YoSubGenerator.CONTENT_SNIPPET} "${contentSnippetName}" "${contentSnippetType}"`;
+                await createRecord(
+                    Tables.CONTENT_SNIPPET,
+                    command,
+                    portalDir,
+                    watcher
+                );
+            }
+        }
+    } catch (error: any) {
+        throw new Error(error);
+    }
+};
+
+async function getContentSnippetInputs(selectedWorkspaceFolder: string) {
+    const contentSnippetTypes: QuickPickItem[] = ["html", "text"].map(
+        (label) => ({ label })
+    );
+
+    interface State {
```

Comment:
interfaces should be in beginning of class - if they are in same file. Or just add a separate file for interfaces